File: tokenize/tokenize.clj
;; From the tokenization tutorial at http://writingcoding.blogspot.com/2008/06/clojure-series-table-of-contents.html

(in-ns 'word)
(clojure/refer 'clojure)

(def stop-words
  #{"a" "in" "that" "for" "was" "is" "it" "the" "of" "and" "to"})
(def token-regex #"\\w+")

(defn to-lower-case [str]
  (. str (toLowerCase)))
(defn tokenize-str 
  ([input-string]
     (map to-lower-case (re-seq token-regex input-string)))
  ([input-string stop-word?]
     (filter (complement stop-word?) (tokenize-str input-string))))

File: jruby/tokenize.rb
module Tokenize
  def self.default_stop_words
    %w{ a in that for was is it the of and to}
  end
  def self.[](str, stop = [])
    stop_words = (stop == true) ? default_stop_words : stop
    str.split(/[^\w]/).map{ |s| s.downcase} - stop_words
  end
end
